core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -127,10 +127,16 @@ static TableMetadata newTableMetadata(Schema schema,
   public static class SnapshotLogEntry implements HistoryEntry {
     private final long timestampMillis;
     private final long snapshotId;
+    private final Integer schemaId;
 
-    SnapshotLogEntry(long timestampMillis, long snapshotId) {
+    SnapshotLogEntry(long timestampMillis, long snapshotId, Integer schemaId) {
```

Comment:
Why add the schema ID to the snapshot log as well as the snapshot? The snapshots in the log are all available in table metadata, so it doesn't seem like there is a benefit to adding it.

Reply:
Thank you for the review! I think I originally added it to history so that we can directly query history entries to get both snapshot ID and schema ID when we do time based time travel queries, but later I didn't end up using it for the utility methods I [used to have](https://github.com/apache/iceberg/pull/2275/commits/bc534a8f9551c4fb12226c2121001b7cf98f2bb1) so I think we can drop it. I'll update the PR to reflect this.

Reply:
The title and description of the PR can be amended to reflect this then.